### DISCUSSION TITLE
Using hyphens in rules

### DISCUSSION THREAD

**Discussion by 3goats** [open-policy-agent/community#489] [Community]
Hi I'm trying perform some additional validation on a JWT. I have a claim within my JWT that looks like this: 

```
"main-application.configuration": "Demo Config"
```

If I then use this rule to validate it: 

```
config := payload[1].main-application.configuration == "Demo Config" 
```

I'm getting an unsafe error:

```
1 error occurred: policy.rego:23: rego_unsafe_var_error: var application is unsafe
```

The problem seems to be the - (hyphen).

Is there a solution for this ?

**Comment by johanfylling** [comment]
Just as how you can use a number ( e.g. `1`) in your reference, you can also use strings to deal with situations where the key contains reserved characters:

```
config := payload[1]["main-application"].configuration == "Demo Config"
```

or even

```
config := payload[1]["main-application.configuration"] == "Demo Config"
```

if `main-application.configuration` is your key.

**Reply by 3goats** [reply]
Perfect that did the trick. Thanks for the prompt response.

**Reply by johanfylling** [reply]
Happy to help :)
